### PR TITLE
ENH: Better indexed inputs and output support in itk::TransformixFilter

### PR DESCRIPTION
**Pull request by thewtex**
Improvements to help with functional Python interface.

GetOutput has indexed interface.

"MovingImage" is the required, first indexed input
("TransformParameterObject") is the second.

### CODE REVIEW

File: Testing/itkTransformixFilterTest.cxx [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+  ImageType::Pointer resultImage = transformix->GetOutput();
+
+  using ImageWriterType = itk::ImageFileWriter<ImageType>;
+  auto imageWriter = ImageWriterType::New();
+  imageWriter->SetFileName(transformedImageFile);
+  imageWriter->SetInput(resultImage);
+  imageWriter->Update();
+
+  /** Return a value. */
+  return 0;
```
Comment (dzenanz) [inline Testing/itkTransformixFilterTest.cxx:72]: Why not `EXIT_SUCCESS`? Is elastix style `return 0`?

**Review by dzenanz — APPROVED** [review]

File: Core/Main/transformixlib.h [hunk 2]
```diff
@@ -85,4 +85,4 @@ class ELASTIXLIB_API TRANSFORMIX
 
 } // namespace transformix
 
-#endif // end #ifndef __transformixlib_h
+#endif // end#ifndef __transformixlib_h
```
[the remark this answers is not in the document]
Comment (thewtex) [reply Core/Main/transformixlib.h:88]: Reverted

**Review by dzenanz — APPROVED** [review]

**Comment by ViktorvdValk** [comment]
> Improvements to help with functional Python interface.
> 
> GetOutput has indexed interface.
> 
> "MovingImage" is the required, second indexed input
> ("TransformParameterObject") is the first.

This is the other way around right?

**Review by ViktorvdValk — APPROVED** [review]
Nice!
I can't comment on the c++ code, but the index inputs and outputs work in the wheels in itkelastix that are create with this commit.

**Comment by thewtex** [comment]
> This is the other way around right?

Correct! I will update the PR description.

**Pull request merged**